### PR TITLE
Fix offset

### PR DESCRIPTION
**Pull request by aneshodza**
TICKET-19657
The timer used to display random stuff. Fixed that by adjusting for the server and client time zone differences

### CODE REVIEW

**Comment by Janis-Leuenberger** [comment]
Can you explain the original and new behaviour in a bit more detail?

File: app/views/timer_sessions/_timer_container.html.slim [hunk 1]
```diff
@@ -1,4 +1,7 @@
-.box[data-controller="form timer"]
+.box[data-controller="form timer",
+  data-timer-timezone-value="#{Time.zone.now.
+  in_time_zone(@current_user.preference.time_zone).utc_offset / 1.hour}"]
```
Comment (coorasse) [inline app/views/timer_sessions/_timer_container.html.slim:3]: this to me seems too much logic to stay in the view untested. I suggest to move it into an helper and add a test for it

**Review by coorasse — CHANGES_REQUESTED** [review]

File: test/functional/timer_sessions_controller_test.rb [hunk 2]
```diff
@@ -45,6 +45,13 @@ def setup
     assert response.body.include?(@timer_session.splittable_hours.round(2).to_s)
   end
 
+  test 'index with different server time zone' do
+    @user.preference.update(time_zone: 'Tijuana')
+    get(:index)
+    assert_response 200
+    assert response.body.include?('-8')
```
Comment (schmijos) [inline test/functional/timer_sessions_controller_test.rb:52]: That's a bit minimalistic, no?

**Review by schmijos — APPROVED** [review]

File: test/functional/timer_sessions_controller_test.rb (hunk 2 above)
Comment (aneshodza) [reply test/functional/timer_sessions_controller_test.rb:52]: @schmijos I thought I'll write a spec that is influenced by only the feature I want to test.

Comment (aneshodza) [reply test/functional/timer_sessions_controller_test.rb:52]: Agreed. Checking for `assert response.body.include?('data-timer-timezone-value="-8"')` makes more sense

**Pull request merged**